react-playground/src/components/Header/Header.tsx

```diff
@@ -1,51 +1,47 @@
-import { FC, useState } from "react";
+import { FC, useState, useLayoutEffect, useRef, useCallback } from "react";
 import { Link } from "react-router-dom";
 import MainLogo from "../../images/main-logo.svg";
 import MenuIcon from "../../images/menu-icon.svg";
-import { HeaderStyled, DropdownStyled, MenuItemStyled } from "./Header.styles";
-
-const menu = [
-  { name: "Home", route: "/" },
-  { name: "My Results", route: "/results" },
-  { name: "Health profile", route: "/healthprofile" },
-  { name: "Community", route: "/community" },
-  { name: "Resources", route: "/resources" },
-  { name: "Settings", route: "/settings" },
-  { name: "Log out", route: "/logout" },
-];
+import { HeaderStyled, MenuIconStyled } from "./Header.styles";
+import { Menu } from "../Menu";
 
 const Header: FC = () => {
   const [isOpen, setIsOpen] = useState(false);
   const toggling = () => setIsOpen(!isOpen);
 
-  const onItemClicked = (value: Object) => () => {
+  const onItemClicked = () => {
```

Comment:
Since the function only will ever close the menu, this function could be called `closeMenu`, then you can pass this into MenuIcon.